src/Microsoft.Azure.NotificationHubs/NamespaceManager.cs

```diff
@@ -217,12 +218,12 @@ public async Task<string> GetVersionInfoAsync()
                     Path = "/$protocol-version"
                 };
 
-            using(var response = await SendAsync((client) => 
+            using(var response = await SendAsync(() => 
             {
-                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri.Uri);
+                var httpRequestMessage = CreateHttpRequest(HttpMethod.Get, requestUri.Uri);
                 httpRequestMessage.Headers.Add("X-PROCESS-AT", "ServiceBus");
 
-                return client.SendAsync(httpRequestMessage);
+                return httpRequestMessage;
```

Comment:
Why are we returning the request message instead of sending it?

Reply:
We are sending it inside:
`private async Task<HttpResponseMessage> SendAsync(Func<HttpRequestMessage> generateHttpRequestMessage)`